### ISSUE TITLE
exception thrown by BlobContainerClient.exists() after upgrading azure-storage-blob to 12.11.0

### ISSUE THREAD

**Issue by donalmurtagh** [Azure/azure-sdk-for-java#21138]
**Describe the bug**
After upgrading azure-storage-blob from 12.10.2 to 12.11.0 an exception is thrown when I call `BlobContainerClient.exists()` in a test. The test runs against the latest version (3.12.0) of [Azurite](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azurite) as a Docker image

***Exception or Stack Trace***
```
com.azure.storage.blob.models.BlobStorageException: Status code 400, (empty body)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[na:na]
	at com.azure.core.http.rest.RestProxy.instantiateUnexpectedException(RestProxy.java:337) ~[azure-core-1.15.0.jar:na]
	at com.azure.core.http.rest.RestProxy.lambda$ensureExpectedStatus$8(RestProxy.java:393) ~[azure-core-1.15.0.jar:na]
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:44) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.Mono.subscribe(Mono.java:4150) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onComplete(FluxSwitchIfEmpty.java:81) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onComplete(MonoFlatMap.java:181) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.FluxDoFinally$DoFinallySubscriber.onComplete(FluxDoFinally.java:145) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.FluxHandle$HandleSubscriber.onComplete(FluxHandle.java:212) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.FluxMap$MapConditionalSubscriber.onComplete(FluxMap.java:269) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.FluxDoFinally$DoFinallySubscriber.onComplete(FluxDoFinally.java:145) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.FluxHandleFuseable$HandleFuseableSubscriber.onNext(FluxHandleFuseable.java:201) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onNext(FluxContextWrite.java:107) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1815) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.MonoCollectList$MonoCollectListSubscriber.onComplete(MonoCollectList.java:128) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.onComplete(FluxPeek.java:259) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.FluxMap$MapSubscriber.onComplete(FluxMap.java:142) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.netty.channel.FluxReceive.onInboundComplete(FluxReceive.java:401) ~[reactor-netty-core-1.0.6.jar:1.0.6]
	at reactor.netty.channel.ChannelOperations.onInboundComplete(ChannelOperations.java:416) ~[reactor-netty-core-1.0.6.jar:1.0.6]
	at reactor.netty.channel.ChannelOperations.terminate(ChannelOperations.java:470) ~[reactor-netty-core-1.0.6.jar:1.0.6]
	at reactor.netty.http.client.HttpClientOperations.onInboundNext(HttpClientOperations.java:685) ~[reactor-netty-http-1.0.6.jar:1.0.6]
	at reactor.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:94) ~[reactor-netty-core-1.0.6.jar:1.0.6]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436) ~[netty-transport-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324) ~[netty-codec-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296) ~[netty-codec-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251) ~[netty-transport-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.channel.kqueue.AbstractKQueueStreamChannel$KQueueStreamUnsafe.readReady(AbstractKQueueStreamChannel.java:544) ~[netty-transport-native-kqueue-4.1.63.Final-osx-x86_64.jar:4.1.63.Final]
	at io.netty.channel.kqueue.AbstractKQueueChannel$AbstractKQueueUnsafe.readReady(AbstractKQueueChannel.java:382) ~[netty-transport-native-kqueue-4.1.63.Final-osx-x86_64.jar:4.1.63.Final]
	at io.netty.channel.kqueue.KQueueEventLoop.processReady(KQueueEventLoop.java:211) ~[netty-transport-native-kqueue-4.1.63.Final-osx-x86_64.jar:4.1.63.Final]
	at io.netty.channel.kqueue.KQueueEventLoop.run(KQueueEventLoop.java:289) ~[netty-transport-native-kqueue-4.1.63.Final-osx-x86_64.jar:4.1.63.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[netty-common-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.63.Final.jar:4.1.63.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.63.Final.jar:4.1.63.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]
	Suppressed: java.lang.Exception: #block terminated with an error
		at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:99) ~[reactor-core-3.4.5.jar:3.4.5]
		at reactor.core.publisher.Mono.block(Mono.java:1703) ~[reactor-core-3.4.5.jar:3.4.5]
		at com.azure.storage.common.implementation.StorageImplUtils.blockWithOptionalTimeout(StorageImplUtils.java:128) ~[azure-storage-common-12.11.0.jar:na]
		at com.azure.storage.blob.BlobContainerClient.existsWithResponse(BlobContainerClient.java:228) ~[azure-storage-blob-12.11.0.jar:na]
		at com.azure.storage.blob.BlobContainerClient.exists(BlobContainerClient.java:211) ~[azure-storage-blob-12.11.0.jar:na]
```

**To Reproduce**
Run the test below replacing the connection string and container name with the a valid connection string and the name of a container that exists

***Code Snippet***
```java
    @Test
    public void testContainerExists() {
        var client = new BlobServiceClientBuilder().connectionString("CONNECTION STRING").buildClient();
        var blobContainerClient = client.getBlobContainerClient("CONTAINER NAME");
        assertTrue(blobContainerClient.exists());
    }
```

**Expected behavior**
The test should pass with v.12.11.0, because it passes with v.12.10.2

**Screenshots**

**Setup (please complete the following information):**
 - OS: Mac OX 10.15.7
 - IDE: IntelliJ
 - Version of the Library used 12.11.0

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [Y] Bug Description Added
- [Y] Repro Steps Added
- [Y] Setup information Added

**Comment by gapra-msft** [comment]
Hi @donalmurtagh Thank you for reporting this issue. We will take a look at what might be going wrong as soon as we can.

**Comment by gapra-msft** [comment]
Hi @donalmurtagh Thank  you for your patience while we look into this issue. I did a little digging and it looks like the SDK is sending a malformed URL when there is a custom endpoint. 

Looping in @alzimmermsft @anuchandy @srnagar 

I investigated a little bit and I found that [this](https://github.com/Azure/azure-sdk-for-java/blob/2abb908b586c1be644184414969886788a315978/sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java#L232) line in RestProxy is misbehaving if the path already starts with a /. Could one of you please take a look? The creds below are all public (it's just Azurite creds) so its fine to paste here. 

```
BlobServiceClient client = new BlobServiceClientBuilder()
            .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.HEADERS))
            .connectionString("DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;")
            .buildClient();
        BlobContainerClient blobContainerClient = client.getBlobContainerClient("test2");

        blobContainerClient.create();
```

**Comment by donalmurtagh** [comment]
@gapra-msft In case it helps, here are the values I use in the (failing) test for the connection string and container name

Connection String: `DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;`

Container Name:
`bid-events`

**Comment by gapra-msft** [comment]
This PR should resolve the issue and we will take a dependency on it for our next release
https://github.com/Azure/azure-sdk-for-java/pull/21163

**Comment by gapra-msft** [comment]
Closing this issue for now as I believe it has been resolved . Feel free to reopen it in case you hit the issue after we release. We should be releasing our next beta version in the next few upcoming weeks.

**Comment by donalmurtagh** [comment]
@gapra-msft this appears to be resolved in 12.11.1, many thanks

**Comment by camelmann** [comment]
@gapra-msft seems this issue is happening again in 12.20.2
